tools/sof-ll-timer-check.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+'''Script to analyze performance data in SOF FW log output'''
+
+import sys
+import re
+import json
+from statistics import median
+
+# allowed error margin before error is raised for
+# lower observed performance (1.05 -> 5% difference
+# required to raise error)
+AVG_ERROR_MARGIN = 1.05
+
+max_vals = []
+avg_vals = []
+overruns = 0
+
+f = open(sys.argv[1])
+
+for line in f:
+    m = re.search('.*ll timer avg ([0-9]*), max ([0-9]*), overruns ([0-9]*)', line)
+    if m:
+        avg_vals.append(int(m.group(1)))
+        max_vals.append(int(m.group(2)))
+        overruns += int(m.group(3))
+
+median_avg_vals = median(avg_vals)
+print("Measurements:\t\t%d" % len(avg_vals))
+print("Median avg reported:\t%d" % median_avg_vals)
+print("Median max reported:\t%d" % median(max_vals))
+print("Highest max reported:\t%d" % max(max_vals))
+
+if overruns:
+    print("ERROR: %s overruns detected" % overruns, file=sys.stderr)
+    sys.exit(-1)
+
+if len(sys.argv) < 4:
+    print("No reference data for key '%s', unable to check performance against reference")
+    sys.exit(0)
+
+median_avg_ref = None
+dbfile = open(sys.argv[3])
+ref_key = sys.argv[2]
```

Comment:
`argparse` requires a tiny bit more effort than `logging` but it's really worth it. Don't procrastinate. `git grep argparse` will find multiple examples in this repo https://docs.python.org/3/howto/argparse.html